music21/musicxml/m21ToXml.py

```diff
@@ -385,16 +392,28 @@ def parse(self, obj=None):
         if obj is None:
             obj = self.generalObj
         outObj = self.fromGeneralObject(obj)
-        # TODO: set whether to do an additional score copy in submethods.
-        return self.parseWellformedObject(outObj)
+        return self.parseWellformedObject(outObj, makeNotation=makeNotation)
 
-    def parseWellformedObject(self, sc):
+    def parseWellformedObject(self, sc, *, makeNotation: bool = True) -> bytes:
         '''
-        parse an object that has already gone through the
-        `.fromGeneralObject` conversion.  Returns bytes.
+        Parse an object that has usually already gone through the
+        `.fromGeneralObject` conversion, which has produced a copy with
+        :meth:`~music21.stream.Score.makeNotation` run on it.
+
+        `makeNotation=True` (default) runs :meth:`~music21.stream.makeNotation`
+        on each of the parts, this time without making a copy, possibly leaving
+        side effects on `sc`.
+
+        Set `makeNotation=False` to avoid making notation.
+
+        Returns bytes.
         '''
+        if not sc.isWellFormedNotation():
```

Comment:
but I think it would still be better in `fromScore()` -- I don't like to make advanced-feature usage too hand-holding, otherwise people will use it when they shouldn't and then ask more Qs when they should be mastering the basic features first.

Reply:
Ah, I had beginner users/list posters in mind here. Should have been a separate PR, maybe.

Reply:
As in, my hand is dangling over the side of the boat reaching for the hands of `makeNotation=True` users here.

Reply:
Or, I suppose, people who don't start with the User's Guide, i.e. github users in all likelihood, and who jump into the module documentation. ScoreExporter. Great! Let's export a score! Will jump them over all of the packing steps. Given that, this is probably better in ScoreExporter.

Reply:
> but I think it would still be better in fromScore()

Good idea -- so we run it ONLY for makeNotation=True, given that makeNotation=False users probably don't want their output cluttered with warnings if they're doing idiosyncratic things.

Got it, I was just not following this for a minute because of the first comment about speed of the test.